src/firefly/js/visualize/ImageViewerLayout.jsx

```diff
@@ -316,19 +317,43 @@ function makeTileDrawers(pv) {
 
     const plot= primePlot(pv);
     const rootDrawer= (
-        <TileDrawerCanvas opacity={1} plot={plot} plotView={pv} key={'TileDrawer:'+pv.plotId} />
+        <ImageRender opacity={1} plot={plot} plotView={pv} key={'TileDrawer:'+pv.plotId} />
     );
     const drawers= pv.overlayPlotViews.filter( (opv) => opv.visible && opv.plot).map( (opv) => {
         return (
-            <TileDrawerCanvas opacity={opv.opacity} plot={opv.plot} plotView={pv}
-                key={'TileDrawer-overlay:'+opv.imageOverlayId}
+            <ImageRender opacity={opv.opacity} plot={opv.plot} plotView={pv}
+                         tileAttributes={opv.colorAttributes}
+                         shouldProcess={(im, newData, imState, nextImState) => {
+                                  if (newData) return imState.color!== imState.srcImageColor;
+                                  else         return imState.color!==nextImState.color;
+                              }}
+                         processor={makeMaskColorProcessor(opv.colorAttributes.color)}
+                         key={'TileDrawer-overlay:'+opv.imageOverlayId}
             />
         );
     });
     drawers.unshift(rootDrawer);
     return drawers;
 }
 
+
+function makeMaskColorProcessor(colorStr) {
+    return (imageData) => {
+        const cAry= Color.toRGB(colorStr);
+        const {data}= imageData;
+        const len= data.length;
+        for(let i= 0; i<len; i+=4) {
+            if (data[i] || data[i+1] || data[i+2]) {
```

Comment:
Thanks for making me really think about this. You found a bug, if the user every turns it black then the mask gets stuck at that color.

It should be 
`if (data[i] || data[i+1] || data[i+2] || data[i+3]) { `

However, now that I think about it, in the mask case the only element in the array that matters for detecting if I need to set the color is `data[i+3]` so I can optimize it to

` if (data[i+3]) {`

In the array `data[i], data[i+1], data[i+2]` is a RGB and `data[i+3]` is opacity. The images from the server set all 4 to zero because we want full transparency (no mask bit) and `data[i+3]` to 255 if there is a mask bit. So I can just look at the opacity index in the `if`, which should be more efficient. The image arrays are big so efficiency matters.


## _Thank you!_